gui/src/api/DashboardService.tsx

```diff
@@ -180,3 +180,12 @@ export const rebuildDesignStory = (
 export const updateReviewViewedStatus = (story_id: number) => {
   return api.put(`/stories/design/review_viewed/${story_id}`, {});
 };
+
+export const getUserDetails = async (): Promise<UserData> => {
```

Comment:
can you make the function calling this async instead and manage retrieval and error handling there

Reply:
This function is async and the function calling this is async. I'll add error handling